### PR TITLE
Add Palau holidays

### LINKED ISSUE #1238: Add Palau holidays
Links:
  - https://en.wikipedia.org/wiki/Public_holidays_in_Palau
  - https://www.timeanddate.com/holidays/palau/

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

Add Palau holidays (`ARMED_FORCES`, `HALF_DAY`, **`PUBLIC`**).

Closes #1238.

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

**Comment by PPsyrius** [comment]
>  Version ~3.8.0-0 was not found in the local cache
>  Error: The version '3.8' with architecture 'arm64' was not found for macOS 14.4.1.
>  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json

The CI/CD appears to be broken.

**Comment by KJhellico** [comment]
> The CI/CD appears to be broken.

Yes, I saw this error in my repository earlier today.

**Comment by arkid15r** [comment]
> The CI/CD appears to be broken. @KJhellico @arkid15r

The issue was resolved in #1778

File: holidays/countries/palau.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+        # Youth Day.
+        self._add_observed(self._add_holiday_mar_15("Youth Day"))
+
+        # Senior Citizens Day.
+        self._add_observed(self._add_holiday_may_5("Senior Citizens Day"))
+
+        if 2011 <= self._year <= 2012:
+            # Memorial Day.
+            self._add_observed(self._add_holiday_last_mon_of_may("Memorial Day"))
```
Comment (KJhellico) [inline holidays/countries/palau.py:71]:
```suggestion
            self._add_holiday_last_mon_of_may("Memorial Day")
```

File: holidays/countries/palau.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        if self._year >= 2018:
+            # President's Day.
+            self._add_observed(self._add_holiday_jun_1("President's Day"))
+
+        # Constitution Day.
+        self._add_observed(self._add_holiday_jul_9("Constitution Day"))
+
+        # Labor Day.
+        self._add_observed(self._add_holiday_1st_mon_of_sep("Labor Day"))
```
Comment (KJhellico) [inline holidays/countries/palau.py:81]:
```suggestion
        self._add_holiday_1st_mon_of_sep("Labor Day")
```

File: holidays/countries/palau.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+        if self._year >= 2018:
+            # Independence Day.
+            self._add_observed(self._add_holiday_oct_1("Independence Day"))
+
+        # United Nations Day.
+        self._add_observed(self._add_united_nations_day("United Nations Day"))
+
+        # Thanksgiving Day.
+        self._add_observed(self._add_holiday_4th_thu_of_nov("Thanksgiving Day"))
```
Comment (KJhellico) [inline holidays/countries/palau.py:91]:
```suggestion
        self._add_holiday_4th_thu_of_nov("Thanksgiving Day")
```

File: holidays/countries/palau.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+        if self._year >= 2017:
+            # Family Day.
+            self._add_observed(self._add_holiday_4th_fri_of_nov("Family Day"))
```
Comment (KJhellico) [inline holidays/countries/palau.py:95]:
```suggestion
            self._add_holiday_4th_fri_of_nov("Family Day")
```

File: tests/countries/test_palau.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    def test_memorial_day(self):
+        name = "Memorial Day"
+        dt = (
+            # Last Monday of May.
+            "2011-05-30",
+            "2012-05-28",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertNoHolidayName(name, 2010, 2014)
```
Comment (KJhellico) [inline tests/countries/test_palau.py:65]: Let's make the tests more strict.

```suggestion
        self.assertNoHolidayName(name, range(1981, 2011), range(2013, 2040))
```

File: tests/countries/test_palau.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+            "2018-11-23",
+            "2019-11-22",
+            "2020-11-27",
+            "2021-11-26",
+            "2022-11-25",
+            "2023-11-24",
+            "2024-11-22",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertNoHolidayName(name, 2016)
```
Comment (KJhellico) [inline tests/countries/test_palau.py:81]:
```suggestion
        self.assertNoHolidayName(name, range(1981, 2017))
```

File: tests/countries/test_palau.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+    def test_presidents_day(self):
+        self.assertHoliday(f"{year}-06-01" for year in range(2018, 2077))
```
Comment (KJhellico) [inline tests/countries/test_palau.py:84]:
```suggestion
        name = "President's Day"
        self.assertHolidayName(name, (f"{year}-06-01" for year in range(2018, 2040)))
        self.assertNoHolidayName(name, range(1981, 2018))
```

File: tests/countries/test_palau.py [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+        self.assertNoNonObservedHoliday(
+            "2019-05-30",
+            "2024-05-30",
+            "2025-06-02",
+            "2030-05-30",
+            "2031-06-02",
+        )
+
+    def test_independence_day(self):
+        self.assertHoliday(f"{year}-10-01" for year in range(2018, 2077))
```
Comment (KJhellico) [inline tests/countries/test_palau.py:94]:
```suggestion
        name = "Independence Day"
        self.assertHolidayName(name, (f"{year}-10-01" for year in range(2018, 2040)))
        self.assertNoHolidayName(name, range(1981, 2018))
```

**Review by KJhellico — COMMENTED** [review]
Well done! 👍 Just a few suggestions

**Review by KJhellico — COMMENTED** [review]
LGTM!

File: holidays/countries/palau.py [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        self._add_observed(self._add_new_years_day(name))
+        self._add_observed(self._next_year_new_years_day, name=name, rule=SAT_TO_PREV_FRI)
+
+        # Youth Day.
+        self._add_observed(self._add_holiday_mar_15("Youth Day"))
+
+        # Senior Citizens Day.
+        self._add_observed(self._add_holiday_may_5("Senior Citizens Day"))
+
+        if 2011 <= self._year <= 2012:
```
Comment (arkid15r) [inline holidays/countries/palau.py:69]: This line looks odd to me, consider
```suggestion
        if self._year in {2011, 2012}:
```

File: tests/countries/test_palau.py [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+            ("2020-11-26", "Thanksgiving Day"),
+            ("2020-11-27", "Family Day"),
+            ("2020-12-25", "Christmas Day"),
+        )
+
+    def test_2022(self):
+        # https://www.facebook.com/photo/?fbid=2076492362502852&set=a.175933635892077
+        self.assertHolidays(
+            Palau(years=2022),
+            # ("2021-12-31", "New Year's Day (observed)"),
```
Comment (arkid15r) [inline tests/countries/test_palau.py:132]:
```suggestion
```

**Review by arkid15r — COMMENTED** [review]
Looks good, great contribution!

A couple of comments to address:

**Review by arkid15r — APPROVED** [review]
🇵🇼 LGTM

**Pull request merged**